src/surface.rs

```diff
@@ -258,6 +282,61 @@ impl<D: SurfaceMemoryDescriptor> Surface<D> {
         Ok(status)
     }
 
+    pub fn query_error(
+        &self,
+        status: bindings::VAStatus,
+    ) -> Result<Vec<SurfaceDecodeMBError>, VaError> {
+        let mut raw: *const bindings::VASurfaceDecodeMBErrors = std::ptr::null();
+
+        // Safe because `self` represents a valid VASurface.
+        va_check(unsafe {
+            bindings::vaQuerySurfaceError(
+                self.display.handle(),
+                self.id,
+                status,
```

Comment:
The documentation for `vaQuerySurfaceError` mentions the following:

> VA_STATUS_ERROR_DECODING_ERROR should be passed in as "error_status"

which suggests this is the only valid value for `status`, for some reason. Should we just use that and remove the `status` argument for this method?

Reply:
Let's remove it for now then, and if new parameters arise we will add then using a well-defined enum :+1:

Reply:
Sure. Done 😄